tests/container_disk_test.go

```diff
@@ -174,22 +146,25 @@ var _ = Describe("[rfe_id:588][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 	Describe("[rfe_id:273][crit:medium][vendor:cnv-qe@redhat.com][level:component]Starting multiple VMIs", func() {
 		Context("with ephemeral registry disk", func() {
 			It("[test_id:1465]should success", func() {
-				num := 5
-				vmis := make([]*v1.VirtualMachineInstance, 0, num)
-				objs := make([]runtime.Object, 0, num)
-				for i := 0; i < num; i++ {
-					vmi := tests.NewRandomVMIWithEphemeralDiskAndUserdata(cd.ContainerDiskFor(cd.ContainerDiskCirros), "#!/bin/bash\necho 'hello'\n")
-					// FIXME if we give too much ram, the vmis really boot and eat all our memory (cache?)
-					vmi.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory] = resource.MustParse("1M")
-					obj := tests.RunVMI(vmi, 10)
+				const count = 5
+				var vmis []*v1.VirtualMachineInstance
+				for i := 0; i < count; i++ {
+					// Provide 1Mi of memory to prevent VMIs from actually booting.
+					// We only care about the volume containers inside the virt-launcher Pod.
+					vmi := libvmi.NewCirros(libvmi.WithResourceMemory("1Mi"))
+					vmi, err := virtClient.VirtualMachineInstance(testsuite.GetTestNamespace(vmi)).Create(context.Background(), vmi)
+					Expect(err).ToNot(HaveOccurred())
+					libwait.WaitForSuccessfulVMIStart(vmi)
```

Comment:
Do I understand correctly that the old code did not have a wait at all (and thus had a race condition)?

I am all for waiting for the VMIs to start, but I'd do it after all `count` vmis are created, to save time.

Reply:
No, the old implementation was waiting for the vmi to start inside the `verifyContainerDiskVMI` function.
So for the first vmi 5 times, for the second vmi 4 times, for the third vmi 3 times, and so on... waste of time, even if milliseconds.
I agree anyway, to move this under the second for loop

Reply:
The old code had a wait and since it was using a nested-loop, it was basically progressively waiting for every VMI to become available after it was created. In each iteration of the outer loop a new VMI was added to the list, and during the inner loop the existing VMIs should already have waited on, while it still needs to wait for the VMI that was added last.

By waiting for all VMs to be created first, we know that once we reach the second loop all the VMIs are already running. We can take the wiat out of the first loop, but then I'd add another loop for it. So IMO since it doesn't matter for the verification I'd leave it as is.

Reply:
> We can take the wiat out of the first loop, but then I'd add another loop for it. So IMO since it doesn't matter for the verification I'd leave it as is.

Ok. This improvement can be done in a follow-up PR.

Reply:
My idea was to keep it in the first loop so the second loop would only test the scenario where 5 vmis are already running. But in the end both choices are valid. WDYT, which one do you prefer?

Reply:
If you wait on creation, you serialize the processes which prolongs the test significantly.
If you first create all, then wait for each, it should be much faster.

Reply:
I moved the wait to the second loop.

Reply:
> I moved the wait to the second loop.

Thanks! But since this changes what the test does, I would have preferred if it was in its own commit, explaining why it is more sensible to create all VMIs first and wait for them later. Or at least add a word about it in the second commit.

I'm not really worried about this not-so-important test; but it is a good practice to separate mere cleanup from improvements.

Reply:
In this way, it should actually be more in line with how the test originally worked.